client/homebrew/editor/editor.jsx

```diff
@@ -78,15 +79,53 @@ const Editor = createClass({
 		}, 1);
 	},
 
-	highlightPageLines : function(){
+	highlightCustomMarkdown : function(){
 		if(!this.refs.codeEditor) return;
 		const codeMirror = this.refs.codeEditor.codeMirror;
 
+		//reset custom text styles
+		const customHighlights = codeMirror.getAllMarks();
+		for (let i=0;i<customHighlights.length;i++) customHighlights[i].clear();
```

Comment:
I'm wondering why it is needed in context that we didn't have this reset before - does it mean that we have introduced something new that requires us to make a reset or we had a bug all this time and now it is being fixed?

Reply:
There are several parts to this:
1) codeMirror has two style settings. "line classes" (the whole line is styled, which we currently do to highlight the `\page` section), and "text marks" (which styles a span of text, and we use on the `{{` characters).
2) The current live site only uses "line classes", but you can see the issue hat currently exists. If you type `\page` the code highlights as we expect, but then if you type extra characters `\pageblahblah` the line is still highlighted even though the code is no longer valid. Same thing if you delete or scramble characters `\pablahblahge` because once the line style is applied, it just remains until the line itself is deleted.
3) The above code handles the refresh for the `text marks` (otherwise once a `{{` block is colored, those colors will just hang around forever), but there is also an added block that now refreshes the "line classes" for `\page` and `\column`.

Reply:
Thanks for clarification!